### ISSUE TITLE
Bookmark problem. Text can be entered in bookmarks and may then un-intentionally disappear

### ISSUE THREAD

**Issue by TinyMCE-User** [tinymce/tinymce#1499]
Description of problem:

Text missing.
When calling "getBookmark" tinyMCE inserts a span in the text. 
On later editing new text can be un-intentionally entered into this span.
When using "getContent" the span - and the text - is removed. 
Thus text is missing from the content.
(Found in In IE 11)

Steps to reproduce:
1. Edit at tinyMCE area 
2. Call "getBookmark" - for instance using on("blur")
3. Go back to the editor and continue writing at the last cursor position
4. Call getContent

Expected result:
All text should be there

Actual result:
Text entered after "getBookmark" was called may be removed when tinyMCE is removing the bookmark

> _**Legacy information imported from TinyMCE bug tracker:**_
> 
> _[#T6757](http://archive.tinymce.com/develop/bugtracker_view.php?id=6757) posted by [lewjlewj](http://archive.tinymce.com/profile.php?name=lewjlewj)_
> 
> _**Tags:** [msie]_
> _**Status:** Open_
> _**Resolution:** None_
> _**Attached URL:** none_

**Comment by blackcathikari** [comment]
Hi, we've investigated this issue and cannot reproduce it in TinyMCE v4.4.0. Thanks, the TinyMCE Team.

**Comment by chuckle123** [comment]
Hi, I am experiencing this same issue. Below is the output of the getContent({format: 'raw'}) function. I experienced this issue when calling getContent after each .on('change'...) and .on('keyUp'...) and spamming enter a couple times then typing. One odd thing shown is that there are multiple data-mce-type spans without ids.

What fixed my problem was calling getBookmark with the following parameters...
`var cursorBookmark = tinymce.activeEditor.selection.getBookmark(1, true);`

```
<div>
    <span data-mce-type="bookmark" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
        <span data-mce-type="bookmark" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
            <span data-mce-type="bookmark" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                <span data-mce-type="bookmark" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                    <span data-mce-type="bookmark" id="mce_12_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                        <span data-mce-type="bookmark" id="mce_13_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                        </span>
                        <span data-mce-type="bookmark" id="mce_14_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                        </span>
                        <span data-mce-type="bookmark" id="mce_15_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                        </span>
                    </span>
                    a
                    <span data-mce-type="bookmark" id="mce_17_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                    </span>
                    <span data-mce-type="bookmark" id="mce_16_start" data-mce-style="overflow:hidden;line-height:0px" style="overflow:hidden;line-height:0px">
                    </span>
                </span>
            </span>
        </span>
    </span>
</div>
```

Hope this helps,
Cam

**Comment by matt-in-a-hat** [comment]
I have seen this issue on both v4.7.9 and v5.0.1
By just tracking the caret with this
```js
editor.on('blur', function () {
  try {
    window.currentEditorBookmark = editors.selection.getBookmark();
  } catch (ex) {}
})
```
Sometimes when calling `editor.getContent()` it has chunks of content missing.
Particularly after I insert an image using the built in `image` plugin, and even if I write some text after that, the result of `getContent()` will consistently be missing the image and the text entered straight afterwards.

Using `getBookmark(1, true)` seems to have alleviated the issue for me, but I still think this is a serious bug with the normal `getBookmark` functionality.

**Comment by lnewson** [comment]
I was able to reproduce the issue using the instructions mentioned above, so reopening this so we can investigate more. I believe @chuckle123 might be on to something with the bookmark nodes without any id's, as that shouldn't be happening.

**Comment by lnewson** [comment]
This should now be fixed in 5.0.2 and will also be included in the next 4.x release. Please let us know if you find any further problems!
